src/argsparser.c

```diff
@@ -301,6 +302,20 @@ void parse_recurring_delay(char *arg)
     free(x);
 }
 
+int isnumber(char *pin)
```

Comment:
thanks for your research.
so in that case would it make sense to change the semantics of the function such that it checks:
- whether string is not numeric. if not, one-shot mode.
- if string is numeric and of size 8, check checksum. if checksum mismatch, one-shot mode, else normal operation.
- if string is numeric and of size 4,7, assume normal operation.
- if string is numeric and of any other size, one-shot mode.

?

as for your comment

> without handling possible different situations

:-) well, this should probably have been done from the start, so you can see it as a general improvement. it doesn't make sense to try to advance the counter when the pin at hand doesn't fit the scheme.